pick.c

```diff
@@ -1011,7 +1032,8 @@ get_key(const char **key)
 		 */
 		c = buf[len - 1];
 		while (c < '@' || c > '~')
-			c = tty_getc();
+			if (read(fileno(tty_in), &c, 1) == ERR)
```

Comment:
This will break on Big Endian machines. Since `c` is declared as an `int` the read byte will end up in the Most-Significant-Byte (`c & 0xff000000`). The remaning bytes of `c` will be filled with potential garbage from stack since `c` is not initialized. Reading uninitialized memory from the stack is in addition undefined behavior. `c` should therefore be declared as `char` and preferably `unsigned char`. Note that this is only important when passing the argument to `read(3)`. The variable can then safely be promoted to a larger type such as `int`.

Prior to release, I always run the tests on an old Mac with a 32-bit PPC processor.

Reply:
Thanks for the clarification: I've fixed this. Would it make sense (although not necessary) to change it also in the original code?

Reply:
Not necessary due to `int getc(FILE *)`.